src/Microsoft.Azure.SignalR.AspNet/Transports/AzureTransport.cs

```diff
@@ -96,12 +99,13 @@ private async Task LifetimeExecute()
 
         public Task Send(object value)
         {
-            if (_context.Environment.TryGetValue(AspNetConstants.Context.AzureServiceConnectionKey, out var connection) && connection is IServiceConnection serviceConnection)
-            {
-                // Invoke service connection
-                return serviceConnection.WriteAsync(ConnectionId, value, _serviceProtocol, _serializer, _pool);
+            if (_clientConnectionManager.TryGetClientConnection(ConnectionId, out var connection))
+            {
+                if (connection.ServiceConnection != null)
+                {
+                    return connection.ServiceConnection.WriteAsync(ConnectionId, value, _serviceProtocol, _serializer, _pool);

```

Comment:
>ServiceConnection.WriteAsync [](start = 38, length = 28)

move `ServiceConnection.WriteAsync` method into the ClientConnectionContext, eg.g. `WriteToTransportAsync`? So that we don't need to have this `ServicConnection` property public